operator/src/test/java/oracle/kubernetes/operator/helpers/EventHelperTest.java

```diff
@@ -0,0 +1,408 @@
+    mementos.add(HelmAccessStub.install());
+
+    testSupport.addToPacket(JOB_POD_NAME, jobPodName);
+    testSupport.addDomainPresenceInfo(info);
+    testSupport.defineResources(domain);
+    DomainProcessorTestSetup.defineRequiredResources(testSupport);
+    HelmAccessStub.defineVariable(OPERATOR_NAMESPACE_ENV, OP_NS);
+    HelmAccessStub.defineVariable(OPERATOR_POD_NAME_ENV, OPERATOR_POD_NAME);
+  }
+
+  @After
+  public void tearDown() {
+    mementos.forEach(Memento::revert);
+  }
+
+  @Test
+  public void whenDomainMakeRightCalled_domainProcessingStartedEventCreated() {
+    makeRightOperation.execute();
+
+    assertThat("Event DOMAIN_PROCESSING_STARTED",
+        containsEvent(getEvents(), DOMAIN_PROCESSING_STARTING_EVENT), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void whenDomainMakeRightCalled_domainProcessingStartedEventCreatedWithExpectedNamespace() {
+    makeRightOperation.execute();
+
+    assertThat("Event DOMAIN_PROCESSING_STARTED message",
+        containsEventWithNamespace(getEvents(),
+            DOMAIN_PROCESSING_STARTING_EVENT),
+        is(Boolean.TRUE));
+  }
+
+  @Test
+  public void whenDomainMakeRightCalled_domainProcessingStartedEventCreatedWithExpectedMessage() {
+    makeRightOperation.execute();
+
+    assertThat("Event DOMAIN_PROCESSING_STARTED message",
+        containsEventWithMessage(getEvents(),
+            DOMAIN_PROCESSING_STARTING_EVENT,
+            String.format(DOMAIN_PROCESSING_STARTING_PATTERN, UID)),
+            is(Boolean.TRUE));
+  }
+
+  @Test
+  public void whenDomainMakeRightCalled_domainProcessingStartedEventCreatedWithInvolvedObject()
+      throws Exception {
+    makeRightOperation.execute();
+
+    assertThat("Event involved object",
+        containsEventWithInvolvedObject(getEvents(), DOMAIN_PROCESSING_STARTING_EVENT, UID, NS),
+        is(Boolean.TRUE));
+  }
+
+  @Test
+  public void whenDomainMakeRightCalled_domainProcessingStartedEventCreatedWithReportingComponent()
+      throws Exception {
+    makeRightOperation.execute();
+
+    assertThat("Event reporting component",
+        containsEventWithComponent(getEvents(), DOMAIN_PROCESSING_STARTING_EVENT),
+        is(Boolean.TRUE));
+  }
+
+  @Test
+  public void whenDomainMakeRightCalled_domainProcessingStartedEventCreatedWithReportingInstance()
+      throws Exception {
+    String namespaceFromHelm = NamespaceHelper.getOperatorNamespace();
+
+    testSupport.runSteps(
+        new EventHelper().createEventStep(
+            new EventData(DOMAIN_PROCESSING_STARTING)
+        ));
+
+    assertThat("Operator namespace ",
+        namespaceFromHelm, equalTo(OP_NS));
+
+    assertThat("Event reporting instance",
+        containsEventWithInstance(getEvents(), DOMAIN_PROCESSING_STARTING_EVENT, OPERATOR_POD_NAME),
+        is(Boolean.TRUE));
+  }
+
+  @Test
+  public void whenCreateEventStepCalled_domainProcessingSucceededEventCreated() {
+    testSupport.runSteps(Step.chain(
+        new EventHelper().createEventStep(
```

Comment:
Intellij should be warning you here that you are accessing static methods using an instance. Pay attention to its warnings. Once you have changed this to use the preferred syntax for static methods, you can combine some lines to make the method shorter.